pkg/controller/clusterdeployment/clusterdeployment_controller_test.go

```diff
@@ -2491,10 +2491,11 @@ func testEmptyClusterDeployment() *hivev1.ClusterDeployment {
 			Kind:       "ClusterDeployment",
 		},
 		ObjectMeta: metav1.ObjectMeta{
-			Name:       testName,
-			Namespace:  testNamespace,
-			Finalizers: []string{hivev1.FinalizerDeprovision},
-			UID:        types.UID("1234"),
+			Name:            testName,
+			Namespace:       testNamespace,
+			Finalizers:      []string{hivev1.FinalizerDeprovision},
+			UID:             types.UID("1234"),
+			ResourceVersion: "999",
```

Comment:
Is there a reference somewhere for this `999` thing?

Without context, I'm guessing it must be a magic number telling the server to force a patch/update that would ordinarily bounce 409-ish.

Reply:
Listed as a breaking change in https://github.com/kubernetes-sigs/controller-runtime/releases/tag/v0.8.0

Reply:
I got [clarification](https://coreos.slack.com/archives/CB48XQ4KZ/p1623082745004400) from the author of that change. He confirmed that these `999` changes are not necessary. The "breaking" aspect of that change is that it would cause tests that were previously succeeding (incorrectly) to fail (correctly). So no fixes are necessary unless we run into such failures.

Reply:
Update: We *do* run into failures -- but they're not conflict-ish. Rather, they're due to the fact that the tests use DeepEquals to determine expected vs actual objects, so now that the fake server is initializing ResourceVersion to `999` those comparisons are failing.

Long-term we ought to fix the tests not to rely on explicit values of this field, which is supposed to be opaque. For now, this is a reasonable workaround.

Reply:
Good summary, tests definite are a bit too brittle.